internal/client.go

```diff
@@ -578,6 +581,9 @@ func NewClient(service workflowserviceclient.Interface, domain string, options *
 	if options != nil && options.Authorization != nil {
 		service = auth.NewWorkflowServiceWrapper(service, options.Authorization)
 	}
+	if options.IsolationGroup != "" {
```

Comment:
check for nil, since it's exposed + was allowed before.  plus there's not really any reason to require people to pass `&ClientOptions{}` everywhere.